app/src/main/java/io/homeassistant/companion/android/settings/ssid/SsidSummaryProvider.kt

```diff
@@ -0,0 +1,20 @@
+package io.homeassistant.companion.android.settings.ssid
+
+import androidx.preference.Preference
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.domain.url.UrlUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
+
+class SsidSummaryProvider @Inject constructor(
+    private val urlUseCase: UrlUseCase
+) : Preference.SummaryProvider<Preference> {
```

Comment:
You should be able to reference your actual preference type, not just the generic preference.